### PR TITLE
build: Relocate pkg-config from tizen to top dir

### PR DESCRIPTION
**Pull request by rzr**
Expose pkg-config as not specific to Tizen,
can be used outside packaging by Linux or
downstream using cmake+gitmodules

libtuv-DCO-1.0-Signed-off-by: Philippe Coval p.coval@samsung.com

### CODE REVIEW

**Comment by rzr** [comment]
It is verified with iotjs too:
https://github.com/Samsung/iotjs/pull/1808

**Comment by LaszloLango** [comment]
It it really necessary to move `libtuv.pc.in` into the root directory? Could you give a bit more detail about this issue?

**Comment by rzr** [comment]
Ok, in iotjs's libtuv's cmake file is used and it is looking up for .pc.in which is only copied when building libtuv as tizen package... I can find you a log file if needed

**Comment by LaszloLango** [comment]
I still do not understand this change. Why does not work the copy in `libtuv.spec`?

**Comment by rzr** [comment]
It works for libtuv.spec but not for iotjs.spec,
well we could eventually copy that .pc file also all projects using lib but this is a bad practice.
Also pkgconfig is not tizen specific  it would be also helpful for linux users if shared in sources instead of using tizen's packaging files or duplicating it.

**Comment by akosthekiss** [comment]
My two cents:
- Pkg-config files are not Tizen-specific, so it's indeed not justified to "hide" the .pc.in file in config/tizen/packaging/.
- Moreover, .pc[.in] files are not necessarily packaging-specific. A locally-built ("non-packaged") library can also have a .pc file and install that during its `make install` step to a proper location.
- Actually, IMO, that's how it should happen in an ideal world. I.e., a library with a cmake-based build system should have something like `install(TARGETS ${lib_target} DESTINATION lib)` _and_ `install(FILES ${pc_file} DESTINATION lib/pkgconfig)` in its cmake files. Of course, that "pc_file" should be configured during the build process, e.g., from a .pc.in file.
- To be honest, however, I have rarely seen the .pc.in file in the root of the project dir but more often in something like a config subdir.
- If I'm not mistaken, the problem with the current build system is that `CONFIGURE_FILE` in cmake/option/option_noarch-tizen.cmake looks for `${libname}.pc.in` in the current source directory (which seems to be the project root directory). But moving the .pc.in file to the root is not the only way to solve that problem: `CONFIGURE_FILE(.../${libname}.pc.in` may also be a viable approach.

So, I think I would try something along the following lines:
- Move libtuv.pc.in to config/.
- Update option_noarch-tizen.cmake to look for the file at the right place.
- (Making libtuv use proper cmake-install would probably be too far-fetched for now, so I'd skip that. That was just some thought experiment above.)
- But double check whether libtuv.spec tries to copy the .pc file from the right directory: `cp ./build/%{platform}/%{build_mode}/cmake/libtuv.pc %{buildroot}/%{_libdir}/pkgconfig/%{name}.pc` seems to be a bit suspicious.

**Comment by rzr** [comment]
thx @akosthekiss for confirming my points, I am Ok to relocate in subdir, this should work too

**Comment by rzr** [comment]
Once merged I will share an other "cmake improvement" patch for packaging/installing files the "unix" way.

File: cmake/option/option_noarch-tizen.cmake [hunk 1]
```diff
@@ -22,7 +22,7 @@ SET(exec_prefix "${prefix}/bin")
 SET(includedir "${prefix}/include/${PROJECT_NAME}")
 SET(libdir "${prefix}/lib")
 SET(libname "${PROJECT_NAME}")
-CONFIGURE_FILE(${libname}.pc.in ${libname}.pc @ONLY)
+CONFIGURE_FILE(config/${libname}.pc.in ${libname}.pc @ONLY)
```
Comment (akosthekiss) [inline cmake/option/option_noarch-tizen.cmake]: Does the third `cp` in the `install` step of the spec file work? It seems to be looking for the pc file in a cmake subdir. Does that exist?

Comment (rzr) [reply cmake/option/option_noarch-tizen.cmake]: yes that's true , I've simplified tizen spec file to rely only on cmake install

**Comment by rzr** [comment]
Hacky new year reviewers ;)

File: cmake/option/option_noarch-tizen.cmake (hunk 1 above)
Comment (akosthekiss) [inline cmake/option/option_noarch-tizen.cmake]: Are these package config commands still needed in this file? This logic seems to be duplicated in higher-level cmake/libtuv.cmake .

**Pull request merged**